ansible_3par_docker_plugin/tasks/install_prerequisites_on_all.yml

```diff
@@ -4,6 +4,80 @@
       msg: "The plugin properties file does not have the [DEFAULT] back end"
     when: INVENTORY['DEFAULT'] is not defined
 
+  - name: Initialize the flag for mixed configuration
+    set_fact:
+      set_mixed: ''  
+ 
+  - name: Initialize an empty list for backend drivers
+    set_fact:
+      my_driver_list: []
+  
+  - name: Save driver for each backend in the list 
+    set_fact:
+      my_driver_list:  "{{my_driver_list +[INVENTORY[item]['hpedockerplugin_driver']]}}"
+    with_items:
+      - "{{ INVENTORY.keys()}}"
+    register: my_driver_list  
+    
+  - name:  Initialize valid driver list
+    set_fact: 
+      default_drivers:
+      - hpedockerplugin.hpe.hpe_3par_iscsi.HPE3PARISCSIDriver
+      - hpedockerplugin.hpe.hpe_3par_fc.HPE3PARFCDriver
+      - hpedockerplugin.hpe.hpe_3par_file.HPE3PARFileDriver
+  
+  - name: Fail when backend drivers are incorrect or mispelled 
+    fail:
+      msg: "Driver {{item}} is incorrect"
+    with_items:
+      - "{{my_driver_list}}"
+    when: >-
+        "{{item}}" not in default_drivers       
+     
+  - name: Setting set_mixed flag to true if ISCSI/FC and FileDriver are present
+    set_fact:
+      set_mixed: 'true' 
+    when: >-
+      ('hpedockerplugin.hpe.hpe_3par_file.HPE3PARFileDriver'  in  my_driver_list and 'hpedockerplugin.hpe.hpe_3par_iscsi.HPE3PARISCSIDriver' in my_driver_list) or 
+      ('hpedockerplugin.hpe.hpe_3par_file.HPE3PARFileDriver'  in  my_driver_list and 'hpedockerplugin.hpe.hpe_3par_fc.HPE3PARFCDriver' in  my_driver_list )    
+     
+  - name: Fail when DEFAULT backend driver is block and DEFAULT_FILE backend is not present in mixed configuration
+    fail :
+      msg: "DEFAULT_FILE backend is not present in mixed configuration"
+    with_items:
+      - "{{ INVENTORY.keys()}}"  
+    when:  >-
+      set_mixed ==true and 
+      (INVENTORY['DEFAULT']['hpedockerplugin_driver'] == 'hpedockerplugin.hpe.hpe_3par_iscsi.HPE3PARISCSIDriver' or INVENTORY['DEFAULT']['hpedockerplugin_driver'] == 'hpedockerplugin.hpe.hpe_3par_fc.HPE3PARFCDriver') and 
```

Comment:
Shouldn't we check for defined for INVENTORY['DEFAULT'] here?

Reply:
Not required since we are verifying it at the initial stage itself,